Automattic/wp-calypso
# Stats: Add hourly views

Related to https://github.com/Automattic/red-team/issues/275.

## Proposed Changes

* Add hour interval support for views

## Why are these changes being made?

* Support hourly views for a single date

## Testing Instructions

* Apply `D166038-code` and sandbox `public-api.wordpress.com`
* Open Calypso Live Branch `/stats/hour/jetpack.com`
* Ensure it works okay (the today from the front end is from user timezone or browser, which we'll improve later)
* Open `/stats/hour/jetpack.com?flags=stats%2Fnew-date-filtering&chartStart=2024-11-14&chartEnd=2024-11-14`
* Ensure it doesn't break things regardless the flag is not applied or not

<img width="1206" alt="image" src="https://github.com/user-attachments/assets/43e5f889-ec01-4490-873a-e3fdab52477f">


## Pre-merge Checklist
